apis/Google.Cloud.Diagnostics.AspNet/Google.Cloud.Diagnostics.AspNet/Trace/CloudTrace.cs

```diff
@@ -119,16 +117,10 @@ public static void Initialize(string projectId, HttpApplication application, Tra
         private void BeginRequest(object sender, EventArgs e)
         {
             TraceHeaderContext headerContext = TraceHeaderContextUtils.CreateContext(HttpContext.Current.Request);
-            TraceOptions headerOptions = _headerFactory.CreateOptions(headerContext);
-
             // If the trace header says to trace or if the rate limiter allows tracing continue.
-            if (!headerOptions.ShouldTrace)
+            if (headerContext.ShouldTrace != true || !_rateFactory.CreateOptions().ShouldTrace)
```

Comment:
Actually, thinking about this again, is this right? This will only allow the trace if both `headerContext.ShouldTrace` and `_rateFactory.CreateOptions().ShouldTrace` are true, but that's not what the comment says. Seems like the || should be an && or the comment needs to be updated.